ngraph/core/src/op/squeeze.cpp

```diff
@@ -18,38 +18,60 @@
 using namespace std;
 using namespace ngraph;
 
-NGRAPH_SUPPRESS_DEPRECATED_START
-
 NGRAPH_RTTI_DEFINITION(op::v0::Squeeze, "Squeeze", 0);
 
 op::Squeeze::Squeeze()
-    : FusedOp()
+    : Op()
 {
 }
 
 op::Squeeze::Squeeze(const Output<Node>& data, const Output<Node>& axes)
-    : FusedOp({data, axes})
+    : Op({data, axes})
 {
     constructor_validate_and_infer_types();
 }
 
-void op::Squeeze::pre_validate_and_infer_types()
+op::Squeeze::Squeeze(const Output<Node>& data)
+    : Op({data})
 {
-    auto data = input_value(0);
-    auto axes_node = input_value(1).get_node_shared_ptr();
+    constructor_validate_and_infer_types();
+}
 
+void op::Squeeze::validate_and_infer_types()
+{
+    NGRAPH_OP_SCOPE(v0_Squeeze_validate_and_infer_types);
+    auto data = input_value(0);
     bool data_has_dynamic_rank = data.get_partial_shape().rank().is_dynamic();
     bool data_has_dynamic_shape = data.get_partial_shape().is_dynamic();
 
-    auto axes_constant = get_constant_from_source(axes_node);
+    std::shared_ptr<op::v0::Constant> axes_constant;
+    if (get_input_size() == 1)
+    {
+        // Handling the case when Squeeze op is created with a single input - data.
+        // This way the following code (validation, shape inference) can be used in both cases.
+        axes_constant = make_shared<op::v0::Constant>(element::i64, Shape{0}, vector<int64_t>{});
+    }
+    else
+    {
+        auto axes_node = input_value(1).get_node_shared_ptr();
+        auto axes_pshape = get_input_partial_shape(1);
+        axes_constant = get_constant_from_source(axes_node);
+
+        NODE_VALIDATION_CHECK(this,
+                              axes_pshape.rank().compatible(0) || axes_pshape.rank().compatible(1),
+                              "Second input (axes) should not be of rank higher than 1. Got: ",
+                              axes_pshape.rank().get_length());
+    }
+
     bool axes_is_empty_constant = (axes_constant && axes_constant->get_data_ptr() != nullptr)
                                       ? axes_constant->cast_vector<int64_t>().empty()
                                       : false;
 
     if (data_has_dynamic_rank || !axes_constant || !axes_constant->get_data_ptr() ||
         (data_has_dynamic_shape && axes_is_empty_constant))
     {
-        set_output_type(0, get_input_element_type(0), PartialShape::dynamic());
+        set_output_type(
+            0, get_input_element_type(0), PartialShape::dynamic(data.get_partial_shape().rank()));
```

Comment:
As in the comment about corresponding `type_prop` test, when `!axes_constant` and there is some `1` dims in the data shape I would expect that rank can be smaller. 
Also for `(data_has_dynamic_shape && axes_is_empty_constant)` we can't assume the output rank will be same as data rank.
This can be true only if there is no dim compatible with `1` in the input shape.